src/meshapi/views/forms.py

```diff
@@ -117,24 +133,47 @@ def join_form(request: Request) -> Response:
             {"detail": "Your address could not be validated."}, status=status.HTTP_500_INTERNAL_SERVER_ERROR
         )
 
-    # Check if there's an existing member. Dedupe on email for now.
-    # A member can have multiple install requests
-    existing_members = Member.objects.filter(
-        Q(primary_email_address=r.email)
-        | Q(stripe_email_address=r.email)
-        | Q(additional_email_addresses__contains=[r.email])
+    # Check if there's an existing member. Group members by matching on both email and phone
+    # A member can have multiple install requests, if they move apartments for example
+    existing_member_filter_criteria = []
+    if r.email:
+        existing_member_filter_criteria.append(
+            Q(primary_email_address=r.email)
+            | Q(stripe_email_address=r.email)
+            | Q(additional_email_addresses__contains=[r.email])
+        )
+
+    if formatted_phone_number:
+        existing_member_filter_criteria.append(
+            Q(phone_number=formatted_phone_number) | Q(additional_phone_numbers=[formatted_phone_number])
+        )
+
+    existing_members = list(
+        Member.objects.filter(
+            reduce(
+                operator.or_,
+                existing_member_filter_criteria,
+            )
+        )
     )
+
     join_form_member = (
         existing_members[0]
         if len(existing_members) > 0
         else Member(
-            name=r.first_name + " " + r.last_name,
+            name=join_form_full_name,
             primary_email_address=r.email,
-            phone_number=r.phone,
+            phone_number=formatted_phone_number,
             slack_handle=None,
         )
     )
 
+    if r.email not in join_form_member.all_email_addresses:
+        join_form_member.additional_email_addresses.append(r.email)
+
+    if formatted_phone_number not in join_form_member.all_phone_numbers:
+        join_form_member.additional_phone_numbers.append(formatted_phone_number)
```

Comment:
Just another brick in the data warehouse ✊🏻 😔 🧱

Reply:
Actually tho do we want to treat this "new" contact info as more valuable? That is to say, replace the old primary with this one and slot the old primary into the `additional` field?

Reply:
I think we've discussed this before and decided "no" because of griefing, but I wonder...

Maybe this is a problem for the "dashboard™". I could see a world where we send people texts/emails to verify them and only _then_ do we replace the primary.

Reply:
maybe ask them to contact support if this happens?

Reply:
Discussed this on a call on Thursday, decided that this approach is fine for now, though it does expose a potential griefing opportunity of being able to add bogus email addresses to a member object. For now, this poses little security risk since these emails aren't really used for any kind of automated validation. 

We propose to securely validate identity using email/phone in the future if needed, we will require the use of the primary email address. You could imagine the logic being something like:

1. "Enter your email/phone to validate your account"
2. User enters non-primary email that _is_ in their additional_emails (which might be a bad email placed by an attacker) 
3. "oopsie, looks like that's not your primary email address, we have sent an email to p***********@gmail.com (primary address). Please click the link there to validate your identity"

This way we require the user to use the first email they ever used with us (or whatever an admin edits their primary email to) since this is very unlikely to have been hijacked